### DISCUSSION TITLE
Feature Request: Client Certificates Support

### DISCUSSION THREAD

**Discussion by c0dingDom** [GrakovNe/lissen-android#56] [Ideas]
Hey guys.

I use a nginx reverse proxy to enroll SSL for my audiobookshelf instance. Since it's available over the internet I would like to also request Client Certs, like I do for other public services I host.

Since your app already supports customt headers (which is nice and not very common), I thought you might be open for such ideas. It would be great if you could add a filepicker to also provide a client cert (.p12) for (m)TLS authenticaten or utilize the android truststore since the cert is aleady added there. 

I know I could do VPN etc., but that's not the way I like to play the game.

It's special interest for sure, but maybe you like the idea.

Best regards.

**Comment by GrakovNe** [comment]
I wanted to let you know that the app already supports self-signed certificates, but it works a little differently. All you need to do is add your certificate to the Android TrustStore (CA) and then restart the app. Once that’s done, the app will automatically recognize and trust your certificate without requiring an additional file picker.

This approach simplifies things and ensures seamless integration while maintaining security. Let me know if you encounter any issues or have further suggestions

**Comment by c0dingDom** [comment]
Thanks for your fast response!

Sorry, for the confusion. I am using let's encrypt for regular SSL certs. no problem with self signed certs. 

I meant mTLS in my request. Like this:

- Client connects to server
- Server presents its TLS certificate
- Client verifies the server's certificate
- **Client presents its TLS certificate** 
- Server verifies the client's certificate
- Server grants access

( https://www.cloudflare.com/learning/access-management/what-is-mutual-tls/ )

The bold part above would be the topic.

**Reply by GrakovNe** [reply]
Thank you for the clarification! 

I understand now what you're looking for with **mTLS**—specifically the ability for the app to provide a client certificate for authentication, in addition to verifying the server's certificate. I'll consider how to implement this in a way that works well for everyone. Thanks for the suggestion!

**Reply by c0dingDom** [reply]
Thank you for considering and thanks for your effort with the app.

**Reply by CypRyanCyp** [reply]
Any news on this?

**Reply by GrakovNe** [reply]
@CypRyanCyp Fixed in https://github.com/GrakovNe/lissen-android/pull/388
